### DISCUSSION TITLE
Estimate previous request for emulate progress bar.

### DISCUSSION THREAD

**Discussion by marvelsrp** [TanStack/query#2847] [Q&A]
I want to make a progress bar when data is loaded via 
`useQueries (queryKey, queryFn, {keepPreviousData: true}`
The duration of the request is unknown. However, in the case of refetch, the request has already been made, which means that we can roughly estimate the duration of the current request with the same parameters for updating the data.
To do this, you need to know the duration. Or the start and end date of the request.
Can you please tell me where in react-query you can get this data? Or how to forward the cache to memory.
```
  const queryClient = useQueryClient();
  const queryCache = queryClient.getQueryCache();
  const cachedQuery = queryCache.find(queryKey);
```
  cachedQuery.state: 
![image](https://user-images.githubusercontent.com/731425/139217837-bdd00f48-d1a0-45c7-9d16-9ad4a7da76f3.png)

**Comment by TkDodo** [comment]
have a look at this discussion: https://github.com/tannerlinsley/react-query/discussions/2788#discussioncomment-1522830
